src/features/events/hooks/useEventMutations.ts

```diff
@@ -62,6 +63,12 @@ export default function useEventMutations(
     });
   };
 
+  const changeEventCampaign = (campaignId: number) => {
+    updateEvent({
```

Comment:
You can modify `updateEvent()` to be async and return a promise when it completes, and then pass that on from here.